src/components/annotations/attributes.js

```diff
@@ -12,6 +12,7 @@ var ARROWPATHS = require('./arrow_paths');
 var fontAttrs = require('../../plots/font_attributes');
 var cartesianConstants = require('../../plots/cartesian/constants');
 var templatedArray = require('../../plot_api/plot_template').templatedArray;
+var axisPlaceableObjs = require('../../constants/axis_placeable_objects.js');
```

Comment:
You could drop `.js` from the end of such requires.
Also I think it would be more consistent with our code base, if you keep that file in `annotations` instead of `constants`.
Then for requiring from `image` and `shapes`, you could point to the attribute (using Lib.extendFlat) or requiring the function located in `annotation`.